packages/fxa-auth-server/lib/routes/recovery-key.js

```diff
@@ -133,37 +112,32 @@ module.exports = (log, db, Password, verifierVersion, customs, mailer) => {
           },
         },
       },
-      handler(request) {
+      async handler(request) {
         log.begin('recoveryKeyExists', request);
 
-        const email = request.payload.email;
-        let uid;
+        const { email } = request.payload;
 
+        let uid;
         if (request.auth.credentials) {
           uid = request.auth.credentials.uid;
         }
 
-        return Promise.resolve()
-          .then(() => {
-            if (!uid) {
-              // If not using a sessionToken, an email is required to check
-              // for a recovery key. This occurs when checking from the
-              // password reset page and allows us to redirect the user to either
-              // the regular password reset or account recovery password reset.
-              if (!email) {
-                throw errors.missingRequestParameter('email');
-              }
-
-              return customs
-                .check(request, email, 'recoveryKeyExists')
-                .then(() => db.accountRecord(email))
-                .then(result => (uid = result.uid));
-            }
-
-            // When checking from `/settings` a sessionToken is required and the
-            // request is not rate limited.
-          })
-          .then(() => db.recoveryKeyExists(uid));
+        if (!uid) {
+          // If not using a sessionToken, an email is required to check
+          // for a recovery key. This occurs when checking from the
+          // password reset page and allows us to redirect the user to either
+          // the regular password reset or account recovery password reset.
+          if (!email) {
+            throw errors.missingRequestParameter('email');
+          }
+
+          await customs.check(request, email, 'recoveryKeyExists');
+          ({ uid } = await db.accountRecord(email));
+        }
+
+        // When checking from `/settings` a sessionToken is required and the
+        // request is not rate limited.
+        return db.recoveryKeyExists(uid);
```

Comment:
Doesn't this need to be awaited?

Reply:
If `db.recoveryKeyExists` returns a promise, nope. Only need the await if we depend on the synchronous-like behavior it provides afterwards.